### ISSUE TITLE
Feature idea: Toggle unicode escapes in string literals

### ISSUE THREAD

**Issue by jpolitz** [brownplt/pyret-lang#431]
Rational numbers don't have a canonical representation; 1/2 and 0.5 mean the same thing in Pyret.  At the repl, these get rendered as a toggle-able element that switches between printed forms.

The same is true of strings --- we can render unicode as escapes or as the printed glyphs.  It would be cool to toggle between `"välilyöntiä"` and `"v\u00E4lily\u00F6nti\u00E4"` with a click, and it might actually be a handy way to experiment with unicode.

Thanks @schanzer for bringing this up.

**Comment by blerner** [comment]
Why stop there? I'd be happy to see other escape sequences (\t, \n, others?) be rendered instead of escaped, too...

I'd also recommend that there be some sort of ui signal (highlighting, background color, different font, something) to signify "this is a rendering of data that's different from its torepr and so is not copy-pasteable, should you be so inclined..."

**Comment by jpolitz** [comment]
Good point on `"\n"` -- that makes it hard to test a function that folds and accumulates a string to produce formatted output.

Numbers appear only slightly differently (purple rather than green), and that could use a better cue, I agree.

**Comment by schanzer** [comment]
It would make sense to use the same UI cue for image magnification. Right now, for example, the following images are hard to tell apart:

i.frame(i.circle(50,"solid","red"))
i.circle(1000,"solid","red")

Perhaps what's needed is a small, subtle icon next to the value, indicating that the user can take an action to view it differently. When the user mouses over it, the cursor changes to something descriptive (magnifier for images, recycling icon for nums or escaped strings, etc). That way the user learns to look for the same cue whenever output is unexpected.

**Comment by jpolitz** [comment]
Relevant for #379

**Comment by schanzer** [comment]
Confirmed working in CPO

**Comment by jpolitz** [comment]
I love it when features that I wanted magically appear without me realizing it.

**Comment by blerner** [comment]
There's no UI cue for this yet, though, so I'd like to keep this issue open.  Connected to that, no decision was made about whitespace escapes being toggleable via this UI, too.

**Comment by schanzer** [comment]
Let's keep this issue open re: whitespace (and clarify the title). I'll open a separate issue for UI-toggles between representations -- that's much broader issue that shouldn't be buried under strings.
